src/colors/RGBColor.ts

```diff
@@ -0,0 +1,141 @@
+import { vec3 } from 'gl-matrix';
+import { Color } from './Color';
+
+/**
+ * An RGB representation of a color in a WebGL context.
+ *
+ * The acronym stands for:
+ *   R: `R`ed
+ *   G: `G`reen
+ *   B: `B`lue
+ *
+ * @class RGBColor
+ */
+export class RGBColor implements Color {
+    private red: number;
+    private green: number;
+    private blue: number;
+
+    /**
+     * Instantiate a new RGBColor object.
+     *
+     * @class RGBColor
+     * @constructor
+     * @param {number} red Red numeric color value.
+     * @param {number} green Green numeric color value.
+     * @param {number} blue Blue numeric color value.
+     */
+    private constructor(red: number, green: number, blue: number) {
+        if (red < 0 || red > 255) {
+            throw new RangeError(`red's value (${red}) must be in range [0, 255]`);
+        } else if (green < 0 || green > 255) {
+            throw new RangeError(`green's value (${green}) must be in range [0, 255]`);
+        } else if (blue < 0 || blue > 255) {
+            throw new RangeError(`blue's value (${blue}) must be in range [0, 255]`);
+        }
+
+        this.red = red;
+        this.green = green;
+        this.blue = blue;
+    }
+
+    /**
+     * Instantiates a new RGBColor object from a hex string.
+     *
+     * @class RGBColor
+     * @method fromHex
+     * @param {string} value A hex string of the form `FF0000`.
+     * @returns {RGBColor}
+     */
+    public static fromHex(value: string): RGBColor {
+        if (value.length !== 6) {
+            throw new Error('Please pass in a hexadecimal string (i.e., FF33AA)');
+        }
+
+        // Split the hexadecimal string into segments of length 2
+        const matches = value.match(/.{2}/g);
```

Comment:
Sounds good, I was thinking it would be a good idea to omit the prefix, but you're right - it's a familiar concept for web devs. I'll update the regex 💯